### PR TITLE
Pull request for #94: allow folders to be excluded from search

### PR DESCRIPTION
**Pull request by stanfordquan**
@yuchuan2016 this should be fairly simple.

I added an argument in `collect_builder_logs` called `excluded_dirs` (which is passed by the user when calling `log.end_log` at `atexit` in `template`. This argument is a str or list of str of directories to be excluded when we search for sconscript.log files to collect in the `sconstruct.log`.

Let me know if there's any question! Thanks

### CODE REVIEW

File: gslab_scons/log.py [hunk 1]
```diff
@@ -74,20 +74,30 @@ def log_timestamp(start_time, end_time, filename = 'sconstruct.log'):
...
+        '''
     builder_log_collect = {}
 
     # Store paths to logs in a list, found from platform-specific command line tool 
     rel_parent_dir = os.path.relpath(parent_dir)
     log_name = 'sconscript*.log'
+    excluded_dirs = misc.make_list_if_string(excluded_dirs)
     if misc.is_unix():
-        command = 'find %s -name "%s"' % (rel_parent_dir, log_name)
+        command = 'find %s -name "%s -type f"' % (rel_parent_dir, log_name)
```
Comment (yuchuan2016) [inline gslab_scons/log.py]: I think this should be `'find %s -name "%s" -type f'`.

**Comment by yuchuan2016** [comment]
@stanfordquan , good work! I have no other comment except the above one. I have another releated comment. The current search for `sconscript*.log` is in all sub-directories. Then if a log file in `build` is also sent to `release`, it will be appended to `sconstruct.log` twice. Shall we worry about this?

**Comment by stanfordquan** [comment]
That's a good point @yuchuan2016. I didn't plan on having duplicated logs around a directory, so I haven't thought about this.

In principle I'm ok with this. This "appending" mechanism is mostly for user readability while working on code / debugging, so I'm fine if there's some redundant information. Otherwise I'm not really sure about how we detect which logs are copied and which logs are created new.

**Comment by yuchuan2016** [comment]
@stanfordquan , great. I'm also fine with this.

**Pull request merged**